meshpy/mesh_creation_functions/beam_basic_geometry.py

```diff
@@ -482,3 +483,171 @@ def create_beam_mesh_arc_at_node(
+        Point lying on the rotation axis. Does not need to align with
+        bottom plane of helix.
+    start_point: np.array, list
+        Start point of the helix. Defines the radius.
+    twist_angle: float
+        Twist angle / Pitch of helix between 0 and 2pi.
+    height_helix: float
+        Height of helix. Mutually exclusive with number of turns.
+    turns: float
+        Number of turns. Mutually exclusive with height of helix.
+
+    **kwargs (for all of them look into create_beam_mesh_function)
+    ----
+    n_el: int
+        Number of equally spaced beam elements along the line. Defaults to 1.
+        Mutually exclusive with l_el.
+    l_el: float
+        Desired length of beam elements. This requires the option interval_length
+        to be set. Mutually exclusive with n_el. Be aware, that this length
+        might not be achieved, if the elements are warped after they are
+        created.
+
+    Return
+    ----
+    return_set: GeometryName
+        Set with the 'start' and 'end' node of the line. Also a 'line' set
+        with all nodes of the line.
+    """
+
+    if not 0 < twist_angle < 2 * np.pi:
+        raise ValueError("Twist angle of helix must be between 0 and 2pi!")
+
+    if height_helix is None and turns is None:
+        raise ValueError("Either provide height_helix or turns!")
+    elif height_helix is not None and turns is not None:
+        raise ValueError("Only provide height_helix OR turns!")
+
+    # determine radius of helix
+    axis = np.array(axis) / np.linalg.norm(np.array(axis))
+    origin = axis_point + np.dot(
+        np.dot((np.array(start_point) - np.array(axis_point)), axis), axis
+    )
+    start_point_origin_vec = start_point - origin
+    radius = np.linalg.norm(start_point_origin_vec)
+
+    # return line if radius of helix is 0 or twist angle is np.pi/2
+    if np.isclose(radius, 0) or np.isclose(twist_angle, np.pi / 2):
+        if turns:
+            raise ValueError(
+                "Radius of helix is 0 or twist angle is 90 degrees! "
+                + "Height of helix can not be determined through turns! "
+                + "Either switch to height of helix or change radius!"
+            )
+
+        warnings.warn(
+            "Radius of helix is 0 or twist angle is 90 degrees! "
+            + "Simple line geometry is returned!"
+        )
+
+        return create_beam_mesh_line(
+            mesh,
+            beam_object,
+            material,
+            start_point=start_point,
+            end_point=start_point + height_helix * axis,
+            **kwargs,
+        )
+
+    # generate simple helix
+    if height_helix:
+        end_point = np.array([radius, height_helix / np.tan(twist_angle), height_helix])
+    elif turns:
+        end_point = np.array(
+            [
+                radius,
+                2 * np.pi * radius * turns,
+                2 * np.pi * radius * turns * np.tan(twist_angle),
+            ]
+        )
```

Comment:
Sounds like a very good plan. I'm currently trying your supposed approach and unfortunately I am not able to get it working.

I definitely want to allow the case of `radius == 0`. Therefore, I want to return a straight geometry if the input results in a straight line. If I pursue your supposed approach and create a mesh line and wrap it around a cylinder it results in an error as the `radius_phi` within the `wrap_cylinder` function is zero. The same happens if i define the `radius=0.0` for the `wrap_cylinder` function.

If it's not a problem I would just keep the special case in which the straight line is returned.

Regarding the triads: As far as I understand the triad orientation does not result in any differences down the road (simulation)? Is it a problem if they are just "random"?

Reply:
You are right, the `wrap_cylinder` function will complain for a radius of 0. Then I suggest you leave this as is, this will also resolve the issue with the "undefined" triads.

Regarding your question, if the beams cross-section is double-symmetric the orientation of the triad does not change the results - in the continuous setting, in the discrete setting there might be some (small) differences.

Reply:
Thanks for the explanation regarding the triad orientation! Makes sense now